guides/customization/tailscale.md

```diff
@@ -0,0 +1,278 @@
+- A [SOCKS5 proxy] listening for connections on `localhost:1080`
+- A [HTTP proxy] listening for connections on `localhost:3128`
+
+[tailscale networking]: https://tailscale.com/
+[tailnet (tailscale private network)]: https://tailscale.com/kb/1136/tailnet/
+[coder workspaces]: ../../workspaces/index.md
+[socks5 proxy]: https://en.wikipedia.org/wiki/SOCKS
+[http proxy]: https://en.wikipedia.org/wiki/Proxy_server#Web_proxy_servers
+
+## Limitations
+
+This guide describes how to install Tailscale in a Ubuntu base image using the
+package manager and running it in userspace networking mode. As a consequence:
+
+- The image requires [Container-based Virtual Machine] workspaces, so that
+  systemd can start the tailscale daemon (`tailscaled`)
+- Users will require `sudo` access in order to configure the tailscale tunnel
+- Inbound connections from other devices on the tailnet to your workspace will
+  appear to originate from localhost
+- Outbound connections to other devices on the tailnet will require support for
+  HTTP proxies, or use a wrapper such as `proxychains4`
+- This example applies to Ubuntu 20.04 LTS (Focal Fossa), and you may need to
+  adapt it for compatibility with your preferred base image
+
+Tailscale does not require root access to operate in userspace networking mode,
+and the requirement to use [Container-based Virtual Machine] workspaces applies
+only to the instructions in this guide. [Contact our support team] if you are
+interested in using Tailscale in your Coder workspace without root access.
+
+[contact our support team]: ../../feedback.md
+[container-based virtual machine]: ../../workspaces/cvms.md
+
+## Step 1: Create the Dockerfile
+
+In Coder, developer workspaces are defined by a Dockerfile that contains the
+apps, tools, and dependencies that you need to work on the project.
+
+> See Docker’s [guide to writing Dockerfiles] for more information.
+
+[guide to writing dockerfiles]:
+  https://docs.docker.com/develop/develop-images/dockerfile_best-practices/
+
+To simplify creating and maintaining the image, we recommend structuring your
+source code so that files added or modified in the image match the hierarchy of
+the target files, in a subdirectory called `files`:
+
+```text
+.
+├── Dockerfile
+└── files
+    ├── etc
+    │   ├── apt
+    │   │   ├── preferences.d
+    │   │   │   └── tailscale
+    │   │   └── sources.list.d
+    │   │       └── tailscale.list
+    │   └── systemd
+    │       └── system
+    │           └── tailscaled.service.d
+    │               └── tailscale.conf
+    └── usr
+        └── share
+            └── keyrings
+                └── tailscale.gpg
+```
+
+While it is possible to configure everything directly in a single Dockerfile, we
+recommend using a folder hierarchy, since it makes it easier to create a
+reproducible image and examine the change history for individual files.
+
+Create the folder hierarchy using:
```

Comment:
Wondering if there should also be a command to create the files after the folders, I was slightly confused that I had to create the files, but it wasn't a huge deal.

```sh
touch files/etc/apt/preferences.d/tailscale \
    files/etc/apt/sources.list.d/tailscale.list \
    files/etc/systemd/system/tailscaled.service.d/tailscale.conf
```